tests/unit-tests/test_udev_wrapper.cpp

```diff
@@ -26,11 +26,17 @@
 #include <poll.h>
 #include <sys/sysmacros.h>
 
+#if defined(__has_feature)
+#   if __has_feature(address_sanitizer) // for clang
+#       define __SANITIZE_ADDRESS__ // GCC already sets this
+#   endif
+#endif
+
```

Comment:
Rather than relying on a compiler specific macro with a reserved name and faking it on a different compiler, I'd prefer to deal with this in CMake. Either by supplying a `MIR_...` definition we control, or by having the affected code in a separate file that we can exclude from the build.